Core/Core/Grades/Model/GradeFilterInteractor.swift

```diff
@@ -0,0 +1,81 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public protocol GradeFilterInteractor {
+    var gradingShowAllId: String { get }
+    var selectedGradingId: String? { get }
+    var selectedSortById: Int? { get }
+    var isParentApp: Bool { get }
+    func saveGrading(id: String?)
```

Comment:
Can we rename this to `saveSelectedGradingPeriod` ? `saveGrading` doesn't really tell what this function does.